### PR TITLE
Auth: SSO settings foundations

### PR DESCRIPTION
**Pull request by mgyongyosi**
**What is this feature?**

Foundations for the SSO Settings API which contains the store, the db migration, the skeleton of the service and the skeleton of the API. **Basic functionalities are working (it's not supposed to be complete), but this is just the ground work** for being able to distribute the work. All of the **other things will be implemented in correspondence with the API implementation (adding validation, handling updates, preventing mid-air collisions, etc).**  

The feature toggle is set to only work in Dev mode.

**Why do we need this feature?**

To be able to update SSO settings through an API without restarting Grafana.

**Who is this feature for?**

**Which issue(s) does this PR fix?**:

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by kalleep — APPROVED** [review]
LGTM, skimmed through it after our call and it is a good start

**Review by dmihai — APPROVED** [review]
It looks good, great work! I added just a few comments.

File: pkg/services/ssosettings/ssosettingsimpl/service.go [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+		ssoSettingsApi.RegisterAPIEndpoints()
+	}
+
+	return svc
+}
+
+var _ ssosettings.Service = (*SSOSettingsService)(nil)
+
+func (s *SSOSettingsService) GetForProvider(ctx context.Context, provider string) (*models.SSOSetting, error) {
+	dto, err := s.store.Get(ctx, provider)
```
[the remark this answers is not in the document]
Comment (mgyongyosi) [reply pkg/services/ssosettings/ssosettingsimpl/service.go:59]: Yes, we check it using a middleware: https://github.com/grafana/grafana/pull/77724/files#diff-fa5c264c7d4cd3ce77970e6b1268c72f8d835d4fb900012a4545abaec0d4e27cR51 (in the

File: pkg/services/ssosettings/database/database.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+		return nil, err
+	}
+
+	return result, nil
+}
+
+func (s *SSOSettingsStore) Upsert(ctx context.Context, provider string, data map[string]interface{}) error {
+	err := s.sqlStore.WithDbSession(ctx, func(sess *db.Session) error {
+		var err error
+		found, err := sess.Where("provider = ? AND is_deleted = false", provider).Exist(&models.SSOSetting{})
```
[the remark this answers is not in the document]
Comment (mgyongyosi) [reply pkg/services/ssosettings/database/database.go:76]: Thanks, agree!

File: pkg/services/ssosettings/api/api.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+func (api *Api) listAllProvidersSettings(c *contextmodel.ReqContext) response.Response {
+	providers, err := api.SSOSettingsService.List(c.Req.Context(), c.SignedInUser)
+	if err != nil {
+		return response.Error(500, "Failed to get providers", err)
+	}
+
+	return response.JSON(200, providers)
+}
+
+func (api *Api) getSettingsForProvider(c *contextmodel.ReqContext) response.Response {
```
[the remark this answers is not in the document]
Comment (mgyongyosi) [reply pkg/services/ssosettings/api/api.go:66]: Thanks, done!

**Pull request merged**